File: bindings/dotnet/src/Zlink/Runtime/Sockets/PubSocket.cs
// SPDX-License-Identifier: MPL-2.0


namespace Systems.Zlink;

internal sealed class PubSocket : PublisherSocketBase, IPubSocket
{
    public PubSocket(Context context)
        : base(context, SocketType.Pub)
    {
        Options = new PubSocketOptions(this);
    }

    public new PubSocketOptions Options { get; }

    public void SetRoutingId(RoutingId routingId)
    {
        Kernel.SetOption(SocketOptions.RoutingId, routingId.ToBytes());
    }

    public RoutingId GetRoutingId()
    {
        return RoutingId.From(Kernel.GetOption(SocketOptions.RoutingId));
    }

}

File: bindings/node/src/zlink/runtime/sockets/pub_socket.ts
// SPDX-License-Identifier: MPL-2.0

import { PubSocketOptions } from './socket_options';
import {
  PublisherSocket,
} from './socket_operations';
import type { RuntimeContext as Context } from '../core/context';
import { SocketType as NativeSocketType } from '../../contracts/sockets/socket_constants';

export class PubSocket extends PublisherSocket {
  readonly options: PubSocketOptions;
  constructor(ctx: Context) { super(ctx, NativeSocketType.PUB); this.options = PubSocketOptions.create(this); }
}
